aptos-move/aptos-vm/src/adapter_common.rs

```diff
@@ -199,13 +199,24 @@ pub(crate) fn execute_block_impl<A: VMAdapter, S: StateView>(
             debug!(log_context, "Retry after reconfiguration");
             continue;
         };
-        let (vm_status, output, sender) = adapter.execute_single_transaction(
+        let (vm_status, output_ext, sender) = adapter.execute_single_transaction(
             &txn,
             &data_cache.as_move_resolver(),
             &log_context,
         )?;
-        // TODO: apply deltas.
-        let (_, output) = output.into();
+
+        // Apply deltas.
+        let output = if !output_ext.status().is_discarded() {
```

Comment:
let's hide all this logic inside the output_ext method and unify the API, so it can always return output & status (we can ignore whatever's not needed by _), but internally check if the status is not discarded and apply the deltas.

Reply:
probably don't even need to expose the status() then, even for testing.